### ISSUE TITLE
Javascript assigned-undefined rule returns duplicates

### ISSUE THREAD

**Issue by radekk** [returntocorp/sgrep-rules#1117]
**Describe the bug**

**Rule ID:** `assigned-undefined`

It's reporting the same finding twice because of duplicated pattern statement. This PR remove an additional pattern definition to ensure that only a single line is reported without returning back duplicates.

**To Reproduce**

```shell
→ semgrep --verbose -f semgrep-rules/javascript/lang/best-practice/assigned-undefined.yaml ./project
running 1 rules...
project/foobar.js
severity:warning rule:semgrep-rules.javascript.lang.best-practice.assigned-undefined: `undefined` is not a reserved keyword in Javascript, so this is "valid" Javascript but highly confusing and likely to result in bugs.
49:    options = undefined;
--------------------------------------------------------------------------------
49:    options = undefined;
--------------------------------------------------------------------------------
163:    options = undefined;
--------------------------------------------------------------------------------
163:    options = undefined;
ran 1 rules on 13 files: 4 findings
```

**Expected behavior**

```shell
→ semgrep --verbose -f semgrep-rules/javascript/lang/best-practice/assigned-undefined.yaml ./project
running 1 rules...
project/foobar.js
severity:warning rule:semgrep-rules.javascript.lang.best-practice.assigned-undefined: `undefined` is not a reserved keyword in Javascript, so this is "valid" Javascript but highly confusing and likely to result in bugs.
49:    options = undefined;
--------------------------------------------------------------------------------
163:    options = undefined;
ran 1 rules on 13 files: 2 findings
```

**Comment by radekk** [comment]
Code can be re-used from this PR I created but needed to close: https://github.com/returntocorp/semgrep-rules/pull/1116

**Comment by brendongo** [comment]
Looked into why this was getting past our dedup logic and it's because the patterns they matched differed by the semi-colon so the first match reported until the d in undefined while the second match reported until the ;

Thanks for the find @radekk looks like this might have been a hold over from before we fixed issues on trailing semi-colons way back in 0.8.0

**Comment by radekk** [comment]
Always happy to help, thank you for letting me know 🎉
